pandas/core/frame.py

```diff
@@ -4818,72 +4818,88 @@ def aggregate(self, func, axis=0, *args, **kwargs):
-            DataFrame will always be returned.
-
-            .. deprecated:: 0.23.0
+            `apply` will use `reduce` to determine whether the result
+            should be a Series or a DataFrame. If ``reduce is None`` (the
+            default), `apply`'s return value will be guessed by calling
+            `func` on an empty Series
+            (note: while guessing, exceptions raised by `func` will be
+            ignored).
+            If ``reduce is True`` a Series will always be returned, and if
+            ``reduce is False`` a DataFrame will always be returned.
+
+            .. deprecated:: 0.23.0.
                This argument will be removed in a future version, replaced
-               by result_type='reduce'.
+               by ``result_type='reduce'``.
 
-        result_type : {'expand', 'reduce', 'broadcast, None}
-            These only act when axis=1 {columns}:
+        result_type : {'expand', 'reduce', 'broadcast', `None`}
+            These only act when ``axis=1`` (columns):
 
             * 'expand' : list-like results will be turned into columns.
-            * 'reduce' : return a Series if possible rather than expanding
-              list-like results. This is the opposite to 'expand'.
+            * 'reduce' : returns a Series if possible rather than expanding
+              list-like results. This is the opposite of 'expand'.
             * 'broadcast' : results will be broadcast to the original shape
-              of the frame, the original index & columns will be retained.
+              of the DataFrame, the original index and columns will be
+              retained.
 
-            The default behaviour (None) depends on the return value of the
+            The default behaviour `None` depends on the return value of the
             applied function: list-like results will be returned as a Series
             of those. However if the apply function returns a Series these
             are expanded to columns.
 
-            .. versionadded:: 0.23.0
+            .. versionadded:: 0.23.0.
 
         args : tuple
-            Positional arguments to pass to function in addition to the
-            array/series
-        Additional keyword arguments will be passed as keywords to the function
+            Positional arguments to pass to `func` in addition to the
+            array/series.
+        **kwds
+            Additional keyword arguments to pass as keywords arguments to
+            `func`.
 
         Notes
         -----
-        In the current implementation apply calls func twice on the
+        In the current implementation apply calls `func` twice on the
         first column/row to decide whether it can take a fast or slow
-        code path. This can lead to unexpected behavior if func has
+        code path. This can lead to unexpected behavior if `func` has
         side-effects, as they will take effect twice for the first
         column/row.
 
```

Comment:
Can you get rid of the "We use this DataFrame to illustrate" at the beginning of the examples? I don't think it adds any value.

Also, I'd personally prefer to create a smaller (e.g 3 rows, 2 cols) dataframe from a Python list, instead of all that numpy machinery. :) And if the example is with `sqrt`, I'd use numbers like `9` so the user can quickly see what `apply` does, which is the goal of this example.

Reply:
Yes, it's simpler. See d52ac81e7

Reply:
@datapythonista In commit 3160fb805, I suppress an unnecessary test, unintentionally introduced in previous commit (d52ac81)